modules/crawler.py

```diff
@@ -330,47 +284,39 @@ def get_techstack_indicators(domains: List[str], api_key: str):
         api_result = requests.get(tech_stack_query)
         data = json.loads(api_result.content)
         # API supports querying multiple sites at a time, hence the embedded structure
+        technologies = []
         for result_item in data["Results"]:
             result = result_item["Result"]
-            tech_stack = []
             for path in result["Paths"]:
-                technologies = [
-                    {
-                        "indicator_type": "techstack",
-                        "indicator_content": {
+                technologies.extend([
+                    Indicator("techstack",{
                             "name": tech.get("Name"),
                             "link": tech.get("Link"),
                             "tag": tech.get("Tag"),
                             "subdomain": path["SubDomain"],
-                        },
-                        "domain_name": domain,
-                    }
+                        })
                     for tech in path["Technologies"]
-                ]
-                tech_stack.extend(technologies)
-            return tech_stack
-    except IndexError:
-        print(
+                ])
+        return technologies
+    except IndexError as e:
+        logging.warn(
             "Error hit iterating through results. Have you hit your Builtwith API limit?"
         )
-        traceback.print_exc()
-    except Exception:
-        traceback.print_exc()
-    finally:
-        return []
+        raise e
 
-
-def get_tech_identifiers(domains: List[str], api_key: str, save_matches: bool = False):
+@return_empty_if_fails
+def get_tech_identifiers(domains: list[str], api_key: str, save_matches: bool = False) -> list[Indicator]:
     domain_list = ",".join(domains)
     tech_relation_query = (
         f"https://api.builtwith.com/rv2/api.json?KEY={api_key}&LOOKUP={domain_list}"
     )
     api_result = requests.get(tech_relation_query)
-
+    indicators = []
     try:
         data = json.loads(api_result.content)
         for result_item in data["Relationships"]:
             relations = result_item["Identifiers"]
+            # TODO: is this appropriate to keep?
```

Comment:
note: I think this code is outdated for where the codebase is now. it's not great that it's writing out to a file, and also pandas seems overcomplicated. are we still using this function?